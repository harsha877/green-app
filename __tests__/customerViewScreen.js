import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import CustomerViewScreen from '../src/views/customerViewScreen'

const push = jest.fn();
const route = { params: { isAccesibilityModeOn: false } }

test("customer view Screen Loading", () => {
    render(<CustomerViewScreen navigation={{ push }} route={route} />)

})

test("customer view Screen snapshot", () => {
    let customerViewComponenet = render(<CustomerViewScreen navigation={{ push }} route={route} />).toJSON();
    expect(customerViewComponenet).toMatchSnapshot();
})
describe("customer view Screen Query Testing", () => {
    let customerViewComponent;
    beforeEach(() => {
        customerViewComponent = render(<CustomerViewScreen navigation={{ push }} route={route} />);

    })

    /*test("Testing Title", () => {
        const { getAllByText } = customerViewComponent;
        expect(getAllByText("Customer View page").length).toBe(1);
    })*/

    test("Testing Button Text", () => {
        const { getAllByText } = customerViewComponent;
        expect(getAllByText("Refresh").length).toBe(1);
    })

    test("Testing text Styles", () => {
        const props = customerViewComponent.toJSON();
        console.log(props);
        let Styles = props.children[0].children[0].props.style;
        expect(Styles).toStrictEqual({
            alignItems: "center",
            flexDirection: "column",
            justifyContent: "center",
        });
    })

    test("Testing list Styles", () => {
        const props = customerViewComponent.toJSON();
        let Styles = props.children[0].children[1].props.style;
        expect(Styles).toStrictEqual({
            justifyContent: "space-between",
            flexDirection: "column",
            justifyContent: "center",
            alignItems: "center",
        });
    })

    /*test("Testing text input Styles",()=>{
        const props=customerViewComponent.toJSON();
        let Styles=props.children[0].children[1].children[0].props.style;
        expect(Styles).toStrictEqual({
            margin: 15,
            width: 280,
            borderWidth: 1,
            padding: 0,
            borderRadius: 30,
            paddingVertical: 10,
            paddingHorizontal: 10,
            margin: 20,
            alignSelf: "center",
        });
    })*/

    /*test("Testing list Styles",()=>{
         const props=customerViewComponent.toJSON();
        let Styles=props.children[0].children[2].props.style;
        expect(Styles).toStrictEqual({
            justifyContent: "space-between",
            flexDirection: "column",
            justifyContent: "center",
            alignItems: "center",
            });
     })*/

});