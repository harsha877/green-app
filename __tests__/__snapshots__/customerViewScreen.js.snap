// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customer view Screen snapshot 1`] = `
<RCTScrollView>
  <View>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 20,
          "fontWeight": "600",
          "margin": 20,
          "textAlign": "center",
        }
      }
    >
      Customer View page
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 30,
            "borderWidth": 1,
            "margin": 20,
            "padding": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "width": 280,
          }
        }
      >
         
      </TextInput>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#1ca3ec",
              "borderRadius": 50,
              "margin": 10,
              "padding": 0,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
              "width": 125,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 18,
                "padding": 5,
                "textAlign": "center",
              }
            }
          >
            Refresh
          </Text>
        </View>
      </View>
      <Text />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    />
  </View>
</RCTScrollView>
`;
